src/sections/dataset/dataset-action-buttons/access-dataset-menu/AccessDatasetMenu.tsx

```diff
@@ -20,17 +30,81 @@ export function AccessDatasetMenu({ version, permissions }: AccessDatasetMenuPro
     return <></>
   }
 
+  function formatFileSize(bytes: number): string {
+    if (bytes < 1024) {
+      return `${bytes} B`
+    } else if (bytes < 1024 * 1024) {
+      return `${(bytes / 1024).toFixed(2)} KB`
+    } else if (bytes < 1024 * 1024 * 1024) {
+      return `${(bytes / 1024 / 1024).toFixed(2)} MB`
+    } else {
+      return `${(bytes / 1024 / 1024 / 1024).toFixed(2)} GB`
+    }
+  }
+
+  const handleDownload = (type: FileDownloadSizeMode) => {
+    //TODO: implement download feature
+    console.log(`Downloading ${type} zip file`)
+  }
+  const renderDownloadOptions = (datasetContainsTabularFiles: boolean) => {
```

Comment:
I took some time to understand the code, and I wanted to share some suggestions with you. In React, it's often better to be very clear about when something should or shouldn't appear on the screen, especially with conditional rendering.

Right now, the code uses an array of options (const downloadOptions:), and then it maps the result to some buttons. The programmer needs to understand both an array and a mapping.

But it could be easier to understand. For example, if datasetContainsTabularFiles, we could directly show two buttons, and if not, just show one. The difference is in working with buttons instead of dealing with arrays and a mapping.

I'm always in favor of reusing code, except when it comes to the conditional rendering of HTML elements. I prefer duplicating some code and making it explicit that if a certain condition happens, then we show two buttons and if not, just show one



So this:

```
datasetContainsTabularFiles
      ? [
          {
            key: 'original',
            downloadType: FileDownloadSizeMode.ORIGINAL,
            translationKey: 'datasetActionButtons.accessDataset.downloadOriginalZip'
          },
          {
            key: 'archive',
            downloadType: FileDownloadSizeMode.ARCHIVAL,
            translationKey: 'datasetActionButtons.accessDataset.downloadArchiveZip'
          }
        ]
      : [
          {
            key: 'standard',
            downloadType: FileDownloadSizeMode.ORIGINAL,
            translationKey: 'datasetActionButtons.accessDataset.downloadZip'
          }
        ]
```

would be the same as this:

```
{datasetContainsTabularFiles ? (
          <>
            <DropdownButtonItem onClick={() => handleDownload(FileDownloadSizeMode.ORIGINAL)}>
              {t('datasetActionButtons.accessDataset.downloadOriginalZip')} (
              {formatFileSize(
                fileDownloadSizes?.find(
                  (size) => size.fileDownloadSizeMode === FileDownloadSizeMode.ORIGINAL
                )?.size || 0
              )}
              )
            </DropdownButtonItem>
            <DropdownButtonItem onClick={() => handleDownload(FileDownloadSizeMode.ARCHIVAL)}>
              {t('datasetActionButtons.accessDataset.downloadArchiveZip')} (
              {formatFileSize(
                fileDownloadSizes?.find(
                  (size) => size.fileDownloadSizeMode === FileDownloadSizeMode.ARCHIVAL
                )?.size || 0
              )}
              )
            </DropdownButtonItem>
          </>
        ) : (
          <DropdownButtonItem onClick={() => handleDownload(FileDownloadSizeMode.ORIGINAL)}>
            {t('datasetActionButtons.accessDataset.downloadZip')} (
            {formatFileSize(
              fileDownloadSizes?.find(
                (size) => size.fileDownloadSizeMode === FileDownloadSizeMode.ORIGINAL
              )?.size || 0
            )}
            )
          </DropdownButtonItem>
        )}
```

Let me know what you think!

Reply:
Yes, I see what you mean, the version that renders the two conditions is easier to read!  I updated the logic for this.